### PR TITLE
End of Year: Listened categories design 

### PR DESCRIPTION
**Pull request by ashiagr**
| 📘 Project: #410 |
|:---:|

iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/453

## Description
Adds the design for the "listened categories" story.

* The podcasts cover displayed are determined by getting the list of top categories and reversing it, picking the top 1 podcast for each of them
* Background is picked from the middle podcast in the stack (without any gradient)

## Testing Instructions
1. Set `END_OF_YEAR_ENABLED` feature flag to `true` in `base.gradle`
2. Run the app and make sure you're logged in to an account with a few items on Listening History
3. When the prompt appears, tap "View My 2022"
4. Navigate to the third story
4. ✅ Check the design

## Screenshots or Screencast 

| Phone (Pixel 5) |
| -------- | 
|  |

| Tablet (Nexus 9) |
| -------- | 
| |

P.S. -> Story speed issue has re-surfaced after `StoriesFragment` (bottom sheet fragment) was removed in favor of Compose dialog in [this commit](https://github.com/Automattic/pocket-casts-android/commit/07478904b503f0b7718ff120a72d920f28a6e4d4). I tried to fix in 67bf509e11297d02b28c549b851dc78f67a692d2 but I can still the issue. I'll try to fix it in the next PR.

## Checklist
- \If this is a user-facing change, I have added an entry in CHANGELOG.md
- I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt [hunk 1]
```diff
@@ -499,8 +497,7 @@ class MainActivity :
             var showDialog by rememberSaveable { mutableStateOf(false) }
             if (showDialog) {
                 StoriesPage(
-                    viewModel = storiesViewModel,
-                    showDialog = showDialog,
+                    showDialog = true,
```
Comment (mchowning) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:500]: Do we still need this`showDialog` parameter? It looks like we're always passing `true` now, but maybe that's just temporary?

File: modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/views/stories/StoryListenedCategoriesView.kt [hunk 2]
```diff
@@ -1,36 +1,141 @@
...
+    modifier: Modifier = Modifier,
+) {
+    TextP40(
+        text = stringResource(id = LR.string.end_of_year_story_listened_to_categories_subtitle),
+        textAlign = TextAlign.Center,
+        color = story.tintColor,
+        fontWeight = FontWeight.Bold,
+        modifier = modifier
+            .fillMaxWidth()
+            .alpha(0.8f)
```
Comment (mchowning) [inline modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/views/stories/StoryListenedCategoriesView.kt:138]: With some tint colors, this `alpha` can make the text a bit hard to read (to my eyes at least).

**Review by mchowning — APPROVED** [review]
LGTM! 👍

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt (hunk 1 above)
Comment (ashiagr) [reply app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt]: Not needed anymore, removed in 0ac80aa.

File: modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/views/stories/StoryListenedCategoriesView.kt (hunk 2 above)
Comment (ashiagr) [reply modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/views/stories/StoryListenedCategoriesView.kt:133]: I'll try adding a background gradient to improve this in a future PR. 

I also realized we need to display only three podcast covers. This is fixed in 82c7c52.

**Pull request merged**